api/server/server.go

```diff
@@ -167,60 +175,65 @@ func New(
 			WriteTimeout:      httpConfig.WriteTimeout,
 			IdleTimeout:       httpConfig.IdleTimeout,
 		},
-	}, nil
-}
-
-func (s *server) Dispatch() error {
-	listenAddress := net.JoinHostPort(s.listenHost, s.listenPort)
-	listener, err := net.Listen("tcp", listenAddress)
-	if err != nil {
-		return err
 	}
 
-	ipPort, err := ips.ToIPPort(listener.Addr().String())
-	if err != nil {
-		s.log.Info("HTTP API server listening",
-			zap.String("address", listenAddress),
-		)
-	} else {
-		s.log.Info("HTTP API server listening",
-			zap.String("host", s.listenHost),
-			zap.Uint16("port", ipPort.Port),
-		)
+	// Bind a listener for the API endpoint at server construction to
+	// simplify determination of the URI. Previously a listener was
+	// bound in Dispatch but since that method is called concurrently,
+	// retrieving the URI would require synchronization.
+	if err := s.bindListener(httpsEnabled, httpsCert, httpsKey); err != nil {
```

Comment:
Had an idea to make this function a little easier on the eyes: #1757. What do you think?